### PR TITLE
Github Action publish to npm for libms

### PR DESCRIPTION
**Pull request by nichlaes**
## Type of Change

- [X] New feature
- [ ] Bug fix
- [X] Documentation update
- [X] Refactoring
- [ ] Security patch
- [ ] UI/UX improvement

## Description

Made a reusable workflow for publishing NPM package.

## Testing

Forked the main repo, tested the workflow by pushing to personal Node.js NPM 

## Impact

Requires two new secrets. 
- NPM_GITHUB_TOKEN
- NPM_TOKEN

See `docs/developer/github-actions.md`

**Optional:** Workflow takes package name as input for the given package to publish.

## Checklist

- [x] My code adheres to the coding and style guidelines of the project.
- [ ] I have added tests for all the new code and any changes made to
      existing code.
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -8,8 +8,6 @@ on:
     paths:
       - 'docs/**'
   workflow_dispatch:
-    paths:
```
Comment (prasadtalasila) [inline .github/workflows/docs.yml:11]: Is there an advantage to removing this?

File: docs/developer/github-actions.md [hunk 2]
```diff
@@ -8,7 +8,9 @@ from [docker hub](hub.docker.com):
 | `DOCKERHUB_SCOPE` | Username or organization name on docker hub |
 | `DOCKERHUB_USERNAME` | Username on docker hub |
 | `DOCKERHUB_TOKEN` | API token to publish images to docker hub, with `Read`, `Write` and `Delete` permissions |
+| `NPM_TOKEN` | Token to publish npm package to Nodejs registry. |
+| `NPM_GITHUB_TOKEN` | Token to publish npm package to Github registry |
```
Comment (prasadtalasila) [inline docs/developer/github-actions.md:12]: Is this token necessary?

**Review by prasadtalasila — COMMENTED** [review]
@nichlaes thanks for the PR. Please check the comments.

In addition, you have upgraded the libms dependencies but not incremented the package version number. Please increment it in this PR.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (nichlaes) [reply .github/workflows/docs.yml:11]: There is no advantage in having it. It says its no valid

File: docs/developer/github-actions.md (hunk 2 above)
Comment (nichlaes) [reply docs/developer/github-actions.md]: I removed the NPM_GITHUB_TOKEN, since GITHUB_TOKEN can be used. But NPM_TOKEN is required for node js npm

**Comment by nichlaes** [comment]
@prasadtalasila Here is the link to the npm package https://www.npmjs.com/package/@orbit-ece/libms

**Pull request merged**